### PR TITLE
[DO NOT MERGE][PoC] Prow test job

### PR DESCRIPTION
**Pull request by kargakis**
@stevekuznetsov includes a script that allows us to do batch merges w/o any patching involved in any of the prow components.

PoC:
```
I0629 11:02:40.998287       1 mungegithub.go:60] Running mungers
I0629 11:02:41.698004       1 submit-queue-batch.go:307] merging batch: {master 97d901d8e77fff6e8e1b6889e743fe66e7fbbc8b [{%!s(int=3) 0a590d0c5c54aaea23e9142423887196f6dc4294} {%!s(int=4) bcb00a13b286bc025ab560c3d703fca81d9aa9c8}]}
I0629 11:02:42.370580       1 github.go:2531] Munged 0 modified issues. (0 because of status change)
I0629 11:02:42.370605       1 github.go:257] Made 24 API calls since the last Reset 1.176334 calls/sec
I0629 11:02:42.370680       1 mungegithub.go:76] Sleeping for 18.627597114s
I0629 11:02:42.757507       1 github.go:1952] Merging PR# 3
I0629 11:02:42.757535       1 github.go:2171] Commenting in 3: "Automatic merge from submit-queue (batch tested with PRs 3, 4)"
I0629 11:02:44.222187       1 github.go:1444] PR 3 setting "Submit Queue" Github status to "MERGED! (batch)"
I0629 11:02:44.288969       1 github.go:1952] Merging PR# 4
I0629 11:02:44.288996       1 github.go:2171] Commenting in 4: "Automatic merge from submit-queue (batch tested with PRs 3, 4)"
I0629 11:02:46.805112       1 github.go:1444] PR 4 setting "Submit Queue" Github status to "MERGED! (batch)"
I0629 11:02:55.117971       1 github.go:1444] PR 3 setting "Submit Queue" Github status to "Second github e2e run failed."
```
https://ci.openshift.redhat.com/jenkins/job/test_branch_prow_check/29/
https://github.com/origin-prow/origin/pull/3
https://github.com/origin-prow/origin/pull/4

If we are ok with having the python script and updating our jobs to use it, I will open a separate PR with the changes.

### CODE REVIEW

File: sjb/merge_pull_refs.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#
+# And for multiple pull requests that have been batched:
+#
+# master:97d901d8e77fff6e8e1b6889e743fe66e7fbbc8b,4:bcb00a13b286bc025ab560c3d703fca81d9aa9c8,6:4363gsre7568sfsf346fhg3d703f5ab560c3daa8
+pull_refs = sys.argv[3]
+
+os.chdir(working_dir)
+subprocess.call(["git", "checkout", "-q", base_branch])
+
+for pull_num in [r.split(':')[0] for r in pull_refs.split(',')][1:]:
```
Comment (kargakis) [inline sjb/merge_pull_refs.py]: Exactly the same thing with what they are doing upstream @stevekuznetsov

File: sjb/config/test_cases/test_branch_prow_check.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        - PULL_BASE_REF
+    - type: "script"
+      title: "switch origin to origin-prow"
+      repository: "origin"
+      script: |-
+        git remote set-url origin https://github.com/origin-prow/origin
+        git fetch origin
+        git checkout master
+        git reset --hard origin/master
+        wget https://gist.githubusercontent.com/kargakis/d67aafb8700f500b5c2e316a10dc65c7/raw/3bf4032dc3ac2866e61c85a2798f877059f9c00f/merge_pull_refs.py
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_prow_check.yml]: Why are we ignoring the SHA here?

File: sjb/config/test_cases/test_branch_prow_check.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        chmod +x merge_pull_refs.py
+        ./merge_pull_refs.py . "${PULL_BASE_REF}" "${PULL_REFS}"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_branch_prow_check.yml]: Really would love to keep using `oct sync`

File: sjb/merge_pull_refs.py (hunk 1 above)
Comment (stevekuznetsov) [inline sjb/merge_pull_refs.py]: Oh I see, so they disregard the SHA in the pull?

**Comment by kargakis** [comment]
Closing in favor of https://github.com/openshift/aos-cd-jobs/pull/392

**Pull request closed without merging**